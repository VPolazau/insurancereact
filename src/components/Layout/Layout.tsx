import React from 'react';
import cn from 'classnames';
import { useNavigate } from 'react-router-dom';

import { Typography } from '../Typography';
import { Tags } from '../Tags';
import { LeftArrow } from '../../assets/icons';

import styles from './styles.module.css';

interface ILayout {
    title: string;
    isBtnBack?: boolean;
    tags?: { tag: string; path: string; count?: number }[] | null;
    className?: string;
    children?: React.ReactNode;
}

export const Layout: React.FC<ILayout> = ({ title, isBtnBack = false, className, children, tags }) => {
    const navigate = useNavigate();
    let minHeight = '98px';

    const handleBtnBack = () => {
        navigate(-1);
    };

    if (isBtnBack || tags) minHeight = '124px';

    return (
        <div className={styles.content}>
            <div style={{ minHeight }} className={styles.header}>
                <Typography variant="h2">{title}</Typography>
                {isBtnBack && (
                    <button onClick={handleBtnBack} className={styles.btnBack}>
                        <LeftArrow viewBox="0 0 8 14" className={styles.iconArrow} />
                    </button>
                )}
                {tags && (
                    <div className={styles.tagsList}>
                        <Tags list={tags} />
                    </div>
                )}
            </div>
            <div className={cn(styles.body, className)} data-testid="LayoutContent">
                {children}
            </div>
        </div>
    );
};
