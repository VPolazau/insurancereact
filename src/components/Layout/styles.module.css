.content {
    height: calc(100% - 76px);
    display: flex;
    flex-direction: column;
}

.iconArrow {
    height: 14px;
    width: 8px;
}

.header {
    padding: 24px 32px 0;
    background: #FCFFFE;
}

.btnBack {
    display: flex;
    justify-content: center;
    align-items: center;
    background: #FCFFFE;
    padding: 8px 12px 8px 10px;
    border: 1px solid #22C062;
    border-radius: 4px;
    margin-top: 14px;
}

.btnBack:hover {
    background: #E6FEEF;
}

.body {
    height: 100vh;
    background-color: #ECEEED;
    padding: 32px;
    overflow-x: hidden;
    overflow-y: auto;
}

.tagsList {
    display: inline-flex;
    align-items: flex-start;
    gap: 40px;
    margin-top: 12px;
}