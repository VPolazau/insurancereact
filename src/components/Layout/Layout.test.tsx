import { render, screen } from '@testing-library/react';

import { MemoryRouter } from 'react-router-dom';

import { Layout } from './Layout';

describe('Layout', () => {
    it('for property page with mock component', () => {
        render(
            <MemoryRouter>
                <Layout
                    className="layoutClass"
                    title="Имущество"
                    isBtnBack={true}
                    tags={null}
                    children={<div>Some mock component</div>}
                />
            </MemoryRouter>
        );
        const layout = screen.getByTestId('LayoutContent');

        expect(layout.textContent).toBe('Some mock component');
        expect(layout.classList.contains('layoutClass')).toBe(true);
        expect(screen.getByText('Имущество')).toBeInTheDocument();
        expect(layout).toBeInTheDocument();
        expect(layout).toMatchSnapshot();
    });
});
