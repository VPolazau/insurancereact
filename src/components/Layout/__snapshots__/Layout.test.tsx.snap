// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout for property page with mock component 1`] = `
<div
  class="body layoutClass"
  data-testid="LayoutContent"
>
  <div>
    Some mock component
  </div>
</div>
`;
